pytorch_pretrained_bert/modeling_gpt2.py

```diff
@@ -557,16 +558,16 @@ def forward(self, input_ids, position_ids=None, token_type_ids=None, past=None,
 
         output_shape = input_shape + (hidden_states.size(-1),)
 
-        presents = []
+        presents = ()
         all_attentions = []
-        all_hidden_states = []
+        all_hidden_states = ()
         for i, (block, layer_past) in enumerate(zip(self.h, past)):
             if self.output_hidden_states:
-                all_hidden_states.append(hidden_states.view(*output_shape))
+                all_hidden_states += (hidden_states.view(*output_shape),)
```

Comment:
Better to very explicitly say `all_hidden_states = all_hidden_states + (hidden_states.view(*output_shape),)` so we are sure the states are added at the end.